src/ucp/core/ucp_worker.c

```diff
@@ -1169,6 +1169,76 @@ void ucp_worker_iface_cleanup(ucp_worker_iface_t *wiface)
     ucp_worker_uct_iface_close(wiface);
 }
 
+static ucs_status_t ucp_worker_add_resource_cms(ucp_worker_h worker)
+{
+    ucp_context_h   context  = worker->context;
+    ucs_status_t    status;
+    ucp_rsc_index_t i, j;
+
+    if (!ucp_worker_sockaddr_is_cm_proto(worker)) {
+        return UCS_ERR_UNSUPPORTED;
+    }
+
+    UCS_ASYNC_BLOCK(&worker->async);
+    worker->num_cms = 0;
+    for (i = 0; i < context->num_cmpts; ++i) {
+        if (context->tl_cmpts[i].attr.flags & UCT_COMPONENT_FLAG_CM) {
+            ++worker->num_cms;
```

Comment:
maybe use a bitmap: context->cm_tls_bitmap
this was the loop in line 1202 could do use_for_each_bit, and avoid the if in line 1203